edx_solutions_api_integration/imports/views.py

```diff
@@ -193,7 +194,19 @@ def _enroll_user(self, data, errors):
         try:
             cohort = add_cohort(course_key, CourseUserGroup.default_cohort_name, CourseCohort.RANDOM)
         except (IntegrityError, ValueError):
-            cohort = get_cohort_by_name(course_key, CourseUserGroup.default_cohort_name)
+            # TODO: Revisit this approach, possibly using an atomic transaction for adding a cohort?
+            # Sometimes we can get a situation where although the Cohort exists, read replicas
+            # haven't synced up yet and our read for new Cohorts or related models fails when it hits those replicas.
+            # So we retry a couple times until it works.
+            tried, cohort = 0, None
+            while cohort is None:
+                try:
+                    time.sleep(1)
+                    cohort = get_cohort_by_name(course_key, CourseUserGroup.default_cohort_name)
```

Comment:
Please do it before the `sleep`. If it was refreshed correctly, this would introduce 1s of unnecessary overhead.

Reply:
Ah, you were faster here :P

Reply:
Yeah, I pushed it literally seconds before your comment :p